### PR TITLE
test: Enable all remaining TPCH queries

### LINKED ISSUE #258: Enable all tests in CometTPCHQuerySuite
### What is the problem the feature request solves?

We disabled 4 tests ("q1", "q15", "q17", "q18")  in `CometTPCHQuerySuite` before because Comet produces different results than Spark. We should re-enable them after Comet can produce correct results.

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #258.

### CODE REVIEW

File: spark/src/test/scala/org/apache/spark/sql/CometTPCHQuerySuite.scala [hunk 1]
```diff
@@ -76,7 +76,7 @@ class CometTPCHQuerySuite extends QueryTest with CometTPCBase with SQLQueryTestH
     "q20",
     "q21",
     "q22")
-  val disabledTpchQueries: Seq[String] = Seq("q1", "q15", "q17", "q18")
+  val disabledTpchQueries: Seq[String] = Seq()
```
Comment (viirya) [inline spark/src/test/scala/org/apache/spark/sql/CometTPCHQuerySuite.scala:79]: I re-ran these queries locally and they are all passed now. Let's see if we can have them in CI.

**Comment by viirya** [comment]
cc @sunchao @andygrove

**Review by sunchao — APPROVED** [review]
LGTM (pending CI)

**Comment by viirya** [comment]
Merged. Thanks.

**Pull request merged**